### PR TITLE
gh-102500: Document PEP 688

### PR DESCRIPTION
**Pull request by JelleZijlstra**
Companion to #102521.

* Issue: gh-102500

### CODE REVIEW

File: Doc/reference/datamodel.rst [hunk 1]
```diff
@@ -2849,6 +2849,47 @@ a :exc:`TypeError`.
       The specification for the Python ``match`` statement.
 
 
+.. _python-buffer-protocol:
+
+Emulating buffer types
+----------------------
+
+The :ref:`buffer protocol <bufferobjects>` provides a way for Python
+objects to expose efficient access a low-level memory array. This protocol
```
Comment (hugovk) [inline Doc/reference/datamodel.rst]:
```suggestion
objects to expose efficient access to a low-level memory array. This protocol
```

File: Doc/reference/datamodel.rst [hunk 2]
```diff
@@ -2849,6 +2849,47 @@ a :exc:`TypeError`.
...
+is implemented by builtin types such as :class:`bytes` and :class:`memoryview`,
+and third-party libraries may define additional buffer types.
+
+While buffer types are usually implemented in C, it is also possible to
+implement the protocol in Python.
+
+.. method:: object.__buffer__(self, flags)
+
+   Called when a buffer is requested from ``self`` (for example, by the
+   :class:`memoryview` constructor). The ``flags`` argument is an integer
```
Comment (hugovk) [inline Doc/reference/datamodel.rst]: arguments in italics: https://devguide.python.org/documentation/markup/#quick-reference

```suggestion
   Called when a buffer is requested from *self* (for example, by the
   :class:`memoryview` constructor). The *flags* argument is an integer
```

File: Doc/reference/datamodel.rst [hunk 3]
```diff
@@ -2849,6 +2849,47 @@ a :exc:`TypeError`.
...
+   representing the kind of buffer requested, affecting for example whether
+   the returned buffer is read-only or writable. :class:`inspect.BufferFlags`
+   provides a convenient way to interpret the flags. The method must return
+   a :class:`memoryview` object.
+
+.. method:: object.__release_buffer__(self, buffer)
+
+   Called when a buffer is no longer needed. The ``buffer`` argument is a
```
Comment (hugovk) [inline Doc/reference/datamodel.rst]:
```suggestion
   Called when a buffer is no longer needed. The *buffer* argument is a
```

File: Doc/reference/datamodel.rst (hunk 2 above)
Comment (hugovk) [reply Doc/reference/datamodel.rst]: Or remove *flags* as it's now private and meant to be undocumented:

https://github.com/python/cpython/pull/102521#issuecomment-1464730949

Comment (JelleZijlstra) [reply Doc/reference/datamodel.rst]: This is the `flags` argument to `__buffer__`, not to memoryview.

(Also, thanks for the feedback! Will apply it later today.)

File: Doc/library/inspect.rst [hunk 4]
```diff
@@ -1575,6 +1575,41 @@ the following flags:
    for any introspection needs.
 
 
+Buffer flags
+------------
+
+The :mod:`inspect` module provides an :class:`enum.IntFlag`
```
Comment (hugovk) [inline Doc/library/inspect.rst]:
```suggestion
The :mod:`inspect` module provides an :class:`enum.IntFlag`.
```

**Review by hugovk — APPROVED** [review]

**Review by hauntsaninja — APPROVED** [review]

**Pull request merged**